contracts/mocks/ERC4626Mock.sol

```diff
@@ -20,3 +19,42 @@ contract ERC4626Mock is ERC4626 {
         _burn(account, amount);
     }
 }
+
+contract ERC4626DecimalMock is ERC4626Mock {
+    using Math for uint256;
+
+    uint8 private immutable _decimals;
+
+    constructor(
+        IERC20Metadata asset,
+        string memory name,
+        string memory symbol,
+        uint8 decimalsOverride
+    ) ERC4626Mock(asset, name, symbol) {
+        _decimals = decimalsOverride;
+    }
+
+    function decimals() public view virtual override returns (uint8) {
+        return _decimals;
+    }
+
+    function _initialConvertToShares(uint256 assets, Math.Rounding rounding)
+        internal
+        view
+        virtual
+        override
+        returns (uint256 shares)
+    {
+        return assets.mulDiv(10**decimals(), 10**super.decimals(), rounding);
```

Comment:
Can we simplify it like this just to keep the test setup simpler

```suggestion
        return assets.mulDiv(10**_decimals, 1e18, rounding);
```

Reply:
I like the fact that it is generic. Maybe people will ask us how to override to keep the old behavior and we can point to that ?

Also, the `1e18` is wrong ... it should be the tokens decimals, which in our test is not 18